R/summary_forest_plot.R

```diff
@@ -0,0 +1,637 @@
+      side.xu <- ceiling(absside)
+    } else {
+      #Check the maximum required print range using predictive interval range
+      absside <- max(abs(ma$predint[, 5:7]), abs(mtc$predint[, 2:4]), na.rm = TRUE)
+      side.xl <- -1 * ceiling(absside)
+      side.xu <- ceiling(absside)
+    }
+  }
+  
+  i.pt <- 0
+  i.tx <- 0
+  for (i in 1:tplot) {
+    
+    par(mar = c(0.3, 0.25, 0.2, 0.25))
+    plot(1:10, seq(-3, 3, len = 10), type = "n", axes = FALSE, ylab = "", xlab = "", xlim = c(side.xl, side.xu))
+    
+    #matrix cells alternate background shading
+    if (shgvector[i] == 1) {
+      rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "grey95")
+    } else {
+      rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "white")
+    }
+    
+    i.pt <- i.pt + 1
+    if (i %% (ntx + 1) == 1) {
+      ## Diagonal
+      if (i %% ntx != 0) {
+        rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "white")
+        text(xpos, 2.0, lstx[i %% ntx], cex = ucex, adj = c(0.5, 0.5), font = 2)
+        text(xpos + 0.6, -0.1, sprintf("Rank = %.0f", mtc$rank[i %% ntx, 3]), cex = ucex * 0.9, adj = 0.5, font = 1)
+      } else { #last box
+        rect(par("usr")[1], par("usr")[3], par("usr")[2], par("usr")[4], col = "white")
+        text(xpos, 2.0, lstx[ntx], cex = ucex, adj = c(0.5, 0), font = 2)   #NOTE difference here for lstx!!
+        text(xpos + 0.6, -0.1, sprintf("Rank = %.0f", mtc$rank[ntx, 3]), cex = ucex * 0.9, adj = 0.5, font = 1)
+      }
+      i.pt <- i.pt - 1
+      
+    } else if (i %% ntx == 0 | i %% ntx > i %/% ntx) {
+      ##Lower triangle
+      axis <- (i %% ntx == 0)
+      
+      if (!bpredd) {
+        mtc.est <- c(mtc$lor[i.pt, 2], mtc$lor[i.pt, 3], mtc$lor[i.pt, 4])
+        pw.est <- c(ma$lor[i.pt, 5], ma$lor[i.pt, 6], ma$lor[i.pt, 7])
+      } else {
+        mtc.est <- c(mtc$lor[i.pt, 2], mtc$lor[i.pt, 3], mtc$lor[i.pt, 4], mtc$predint[i.pt, 2], mtc$predint[i.pt, 4])
+        pw.est <- c(ma$lor[i.pt, 5], ma$lor[i.pt, 6], ma$lor[i.pt, 7], ma$predint[i.pt, 5], ma$predint[i.pt, 7])
+      }
+      singleSFP(mtc.est, pw.est, bpredd, axis, mtc$type)
+    } else if (i %% ntx <= i %/% ntx) {
+      ##Upper triangle
+      i.pt <- i.pt - 1
+      i.tx <- ((i %% ntx) - 1) * ntx + (i %/% ntx + 1) - (sum(seq((i %% ntx))))
+      mtc.or <- c(mtc$or[i.tx, 2], mtc$or[i.tx, 3], mtc$or[i.tx, 4])
+      pw.or <- c(ma$or[i.tx, 5], ma$or[i.tx, 6], ma$or[i.tx, 7])
+      
+      singleest(mtc.or, pw.or, xpos, ucex + 0.1)
+    }
+    
+    ##Draw a box around the mulitple plots
+    if (shgvector[i] == 1) {
+      box(lty = 1, col = 'grey85')
+    } else {
+      box(lty = 1, col = 'grey85')
+    }
+  }
+  
+  #Identify original number of tx in analysis - for use in rankjar function
+  ori.ntx <- length(mtc$rkgram[, 1]) / ntx
+  
+  #Draw graphs along the diagonal
+  rankogram(ntx, mtc$rkgram[, 1])
+  
+  #Insert the graph title and x-axis title
```

Comment:
MetaInsight also does RR Risk Ratio, RD Risk Difference, and SMD Standardised mean difference. Do you know of any specific reason why these outcomes aren't included (which can just be that you found documentation that the plot is limited to OR, HR, and MD)?

Reply:
Added handling for other outcomes